_posts/2020-01-13-serverless-express-api.md

```diff
@@ -0,0 +1,562 @@
+---
+layout: post
+title: Build & Deploy a Serverless Express API to AWS
+date: 2020-01-13
+tags: javascript tutorial serverless sam
+author: Matthew Tyler
+---
+
+# Introduction
+
+Released in 2015, AWS API Gateway allows developers to build and publish APIs that can be consumed by clients over public internet and virtual private cloud networks. At Re:Invent 2019, announced a significant addition to the service, that they have called "HTTP APIs". The HTTP API is a direct response to customers who wanted to build simple HTTP backed API's, who did not need the complete feature set of API Gateway. The new HTTP API service is perfect for hosting lamba backend applications that are written in popular HTTP Frameworks like Express, Flask, .NET etc.
+
+# How is this different from the existing API Gateway
+
+This existing API Gateway has A LOT of features. Someone who wants to build an application that proxies requests to a lambda-hosted web framework probably won't need most of those features. By removing those features, we get a simpler service to work with, at improved performance and reduced cost. AWS estimates those migrating to HTTP API's from API Gateway can expect a 70% cost reduction and faster response times.
+
+# Let's create a hello world app with SAM!
+
+Let's build a very simple guestbook API using Express. The guestbook will be used to record a comment, the name of the person who made the comment, and the time the comment was made. We will add an additional endpoint that can retrieve all the comments that have been made, starting with the latest comment. We will use S3 to store the comments. Note that I could use an RDBMS or NoSQL database for this - but because I only care about exposing the ability to list and create entries, I can get away with simple object storage.
```

Comment:
```We will use S3 to store the comments. Note that I could use an RDBMS or NoSQL database for this - but because I only care about exposing the ability to list and create entries, I can get away with simple object storage.```

The last part of the sentence, `I can get away with simple object storage`, doesn't quite read right to me.  I understand what you mean but something jars about it and feel it might confuse some people.  Can you re-word?

Perhaps it's that you haven't explained what S3 is (a simple file/object storage service).